modules/apps/content-dashboard/content-dashboard-journal-impl/src/main/java/com/liferay/content/dashboard/journal/internal/item/action/provider/CompareVersionsJournalArticleContentDashboardItemVersionActionProvider.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.content.dashboard.journal.internal.item.action.provider;
+
+import com.liferay.content.dashboard.item.action.ContentDashboardItemVersionAction;
+import com.liferay.content.dashboard.item.action.provider.ContentDashboardItemVersionActionProvider;
+import com.liferay.content.dashboard.journal.internal.item.action.CompareVersionsJournalArticleContentDashboardItemVersionAction;
+import com.liferay.journal.model.JournalArticle;
+import com.liferay.journal.service.JournalArticleLocalService;
+import com.liferay.journal.service.JournalArticleService;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.security.permission.PermissionChecker;
+import com.liferay.portal.kernel.security.permission.resource.ModelResourcePermission;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.portal.kernel.workflow.WorkflowConstants;
+
+import javax.servlet.http.HttpServletRequest;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Jürgen Kappler
+ */
+@Component(service = ContentDashboardItemVersionActionProvider.class)
+public class
+	CompareVersionsJournalArticleContentDashboardItemVersionActionProvider
+		implements ContentDashboardItemVersionActionProvider<JournalArticle> {
+
+	@Override
+	public ContentDashboardItemVersionAction
+		getContentDashboardItemVersionAction(
+			JournalArticle journalArticle,
+			HttpServletRequest httpServletRequest) {
+
+		if (!isShow(journalArticle, httpServletRequest)) {
+			return null;
+		}
+
+		return new CompareVersionsJournalArticleContentDashboardItemVersionAction(
+			httpServletRequest, journalArticle,
+			_getLatestApprovedJournalArticle(journalArticle), _language);
+	}
+
+	@Override
+	public boolean isShow(
+		JournalArticle journalArticle, HttpServletRequest httpServletRequest) {
+
+		ThemeDisplay themeDisplay =
+			(ThemeDisplay)httpServletRequest.getAttribute(
+				WebKeys.THEME_DISPLAY);
+
+		PermissionChecker permissionChecker =
+			themeDisplay.getPermissionChecker();
+
+		User user = themeDisplay.getUser();
+
+		int countArticles = _journalArticleService.getArticlesCountByArticleId(
```

Comment:
@Stephy6, it's ok to name the variable as "count"

Reply:
Great! thank u!